freqtrade/configuration/config_validation.py

```diff
@@ -207,3 +208,24 @@ def _validate_ask_orderbook(conf: Dict[str, Any]) -> None:
                 "Please use `order_book_top` instead of `order_book_min` and `order_book_max` "
                 "for your `ask_strategy` configuration."
             )
+
+
+def validate_migrated_strategy_settings(conf: Dict[str, Any]) -> None:
+
+    conf = _validate_time_in_force(conf)
+
+
+def _validate_time_in_force(conf: Dict[str, Any]) -> None:
+
+    time_in_force = conf.get('order_time_in_force', {})
+    if 'buy' in time_in_force or 'sell' in time_in_force:
+        if conf.get('trading_mode', TradingMode.SPOT) != TradingMode.SPOT:
+            raise OperationalException(
+                "Please migrate your time_in_force settings to use 'entry' and 'exit'.")
+        else:
+            logger.warning(
+                "DEPRECATED: Using 'buy' and 'sell' for time_in_force is deprecated."
+                "Please migrate your time_in_force settings to use 'entry' and 'exit'."
+            )
+            time_in_force['entry'] = time_in_force.pop('buy')
+            time_in_force['exit'] = time_in_force.pop('sell')
```

Comment:
@samgermain this segment of code is what i always meant when i said "compatibility" code.

It ensures that users simply doing an update on an existing bot will not encounter a "cannot load" situation.
obviously they'll still need to update this over time - but it's not a hard transition that simply breaks a working spot bot.